### PR TITLE
E2E test fixes

### PR DESCRIPTION
**Pull request by JDawes-ScottLogic**
Fixes the majority of failing E2E tests.
Removed unnecessary tests.
Marked tests which may or may not be correct with `[Ignore]` and trait of "E2ERewrite". Examples are the Results link tests, as we need to confirm whether we should have a link or not.

### CODE REVIEW

File: VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs [hunk 1]
```diff
@@ -179,22 +87,26 @@ public void AfterVoting_VoteIsRemembered()
...
+                        IReadOnlyCollection<IWebElement> choices = driver.FindElements(NgBy.Binding("choice.Name"));
+                        choices.First().Click();
+
+                        IWebElement voteButton = FindElementById(driver, "vote-button");
+                        voteButton.Click();
 
                         NavigateBackToVotePage(driver);
 
-                        IWebElement selectedChoice = driver.FindElement(By.CssSelector(".selected-choice"));
+                        IWebElement selectedChoice = driver.FindElement(By.CssSelector(".md-ripple-visible"));
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs]: I know this was done a bit hackily before, but I am wondering if there is a way of doing this that doesn't involve css class selectors.

File: VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs [hunk 2]
```diff
@@ -377,137 +256,146 @@ public static Poll CreatePoll(TestVotingContext testContext)
...
-                    ElectionMode = false
-                };
+                        IWebElement voteButton = FindElementById(driver, "vote-button");
+                        voteButton.Click();
 
-                _context.Polls.Add(_namedBasicPoll);
-                _context.SaveChanges();
+                        Assert.IsFalse(driver.Url.StartsWith(PollResultsUrl));
+                        Assert.IsTrue(driver.Url.StartsWith(PollVoteUrl));
+                    }
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs]: I think either we need a placeholder for an error here, or the vote button needs to not be clickable. (voteButton.Enabled == false)

Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs]: I agree that it needs looking at again. This commit is mainly getting the E2E tests back up and running. Basic Poll has a few issues that want ironing out around the UI appearance.

File: VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs (hunk 1 above)
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs]: Just translating the existing tests so they pass under the new design, but I agree this isn't particularly how we want to be doing it. We could check the model, but then we're not testing the UI display.

**Pull request merged**